addons/mrp/tests/test_bom.py

```diff
@@ -1142,3 +1141,392 @@ def test_replenishment(self):
+        self.assertEqual(bom_from_mo_3.product_id, finished_product)
+        self.assertEqual(bom_from_mo_3.product_qty, 3)
+        self.assertRecordValues(bom_from_mo_3.bom_line_ids, [
+            {'product_id': component_1.id, 'product_qty': 3},
+            {'product_id': component_2.id, 'product_qty': 3},
+        ])
+        # Durations should be divided by 3 (op1: 60 / 3 = 20; op2: 480 / 3 : 160).
+        self.assertRecordValues(bom_from_mo_3.operation_ids, [
+            {'name': "Kiss the Monster on its Forehead 😘", 'workcenter_id': self.workcenter_1.id, 'time_cycle': 20},
+            {'name': "Put the Monster in the Jar!", 'workcenter_id': self.workcenter_2.id, 'time_cycle': 160},
+        ])
+        self.assertEqual(bom_from_mo_3.byproduct_ids.product_id, by_product)
+        self.assertEqual(bom_from_mo_3.byproduct_ids.product_qty, 6)
+
+    def test_bom_updates_mo(self):
+        """ Creates a Manufacturing Order using a BoM, then modifies the BoM.
+        Checks the BoM will be marked as updated in the right situation, and checks the "Update BoM"
+        action update the MO accordingly to the changes done in the BoM.
+        """
+        # Creates a BoM.
+        common_vals = {'type': "product"}
+        finished_product = self.env['product.product'].create(dict(common_vals, name="Monster in Jar"))
+        component_1 = self.env['product.product'].create(dict(common_vals, name="Monster"))
+        component_2 = self.env['product.product'].create(dict(common_vals, name="Jar"))
+        component_3 = self.env['product.product'].create(dict(common_vals, name="Bottle"))
+        by_product = self.env['product.product'].create(dict(common_vals, name="Monster's Tears"))
+        bom = self.env['mrp.bom'].create({
+            'product_tmpl_id': finished_product.product_tmpl_id.id,
+            'product_qty': 1.0,
+            'bom_line_ids': [Command.create({'product_id': p.id, 'product_qty': 1}) for p in [component_1, component_2]],
+        })
+
+        # Creates a MO.
+        mo_form = Form(self.env['mrp.production'])
+        mo_form.bom_id = bom
+        mo_1 = mo_form.save()
+        self.assertEqual(mo_1.move_raw_ids[0].product_uom_qty, 1)
+        self.assertEqual(mo_1.is_outdated_bom, False)
+        # Update MO's component quantity.
+        mo_form = Form(mo_1)
+        with mo_form.move_raw_ids.edit(0) as raw_move:
+            raw_move.product_uom_qty = 123
+        mo_1 = mo_form.save()
+        self.assertEqual(mo_1.move_raw_ids[0].product_uom_qty, 123)
+        self.assertEqual(mo_1.is_outdated_bom, False,
+            "Making a modification in the MO shouldn't mark the BoM as updated")
+
+        # Now, adds an operation and a by-product in the BoM.
+        bom.byproduct_ids = [Command.create({'product_id': by_product.id, 'product_qty': 2})]
+        bom_byproduct = bom.byproduct_ids
+        bom.operation_ids = [Command.create({
+            'name': "Insert gently the Monster in the Jar",
```

Comment:
:crying_cat_face: poor monster

```suggestion
            'name': "Gently insert the Monster in the Jar",
```